test/controllers/api/v1/exchange_rates_controller_test.rb

```diff
@@ -0,0 +1,166 @@
+require "test_helper"
+
+module Api
+  module V1
+    class ExchangeRatesControllerTest < ActionDispatch::IntegrationTest
+      setup do
+        # Crear conjunto de datos de prueba
+        @from = "USD"
+        @to = "EUR"
+        
+        # Fechas para pruebas
+        @today = Date.current
+        @yesterday = 1.day.ago.to_date
+        @last_week = 7.days.ago.to_date
+        
+        # Crear registros de prueba
+        @rate_today = ExchangeRate.create!(
+          from_currency: @from,
+          to_currency: @to,
+          rate: 0.85,
+          date: @today
+        )
+        
+        @rate_yesterday = ExchangeRate.create!(
+          from_currency: @from,
+          to_currency: @to,
+          rate: 0.84,
+          date: @yesterday
+        )
+        
+        @rate_last_week = ExchangeRate.create!(
+          from_currency: @from,
+          to_currency: @to,
+          rate: 0.83,
+          date: @last_week
+        )
+      end
+      
+      test "returns error when required parameters are missing" do
+        get api_v1_exchange_rates_path
+        
+        assert_response :bad_request
+        json = JSON.parse(response.body)
+        assert json["error"].present?
```

Comment:
Has this test passed green? I see the controller setting default values, so it should handle properly any missing parameter.

**The behavior described in the test is inconsistent with the code**.